src/lib.rs

```diff
@@ -10,15 +10,34 @@
 
 extern crate cc3200_sys;
 
+#[macro_use]
 pub mod cc3200;
 pub mod isr_vectors;
 
-// We need to define the panic_fmt "lang item", which is just a function. This specifies
-// what the program should do when a `panic!` occurs. Our program won't panic, so we can leave the
-// function body empty for now.
+// These functions are used by the compiler, but not  are normally provided by libstd.
+#[allow(private_no_mangle_fns)]
 mod lang_items {
+    use core::fmt::Arguments;
+
+    #[lang = "eh_personality"]
+    #[no_mangle]
+    pub extern fn rust_eh_personality() {
+    }
+
+    // This function may be needed based on the compilation target.
+    #[lang = "eh_unwind_resume"]
+    #[no_mangle]
+    pub extern fn rust_eh_unwind_resume() {
+    }
+
     #[lang = "panic_fmt"]
-    extern "C" fn panic_fmt() {}
+    #[no_mangle]
+    pub extern fn rust_begin_panic(_msg: Arguments,
+                                   _file: &'static str,
+                                   _line: u32) -> ! {
+        println!("Panic at {}:{} : {}", _file, _line, _msg);
+        loop {}
```

Comment:
If we're just going to stop the current thread, then perhaps we should kill it using FreeRTOS?

Entering an infinite loop will have a variety of side effects depending on where it happens.

If if happens in a high priority task then it will effectively kill the system.
If it happens in an ISR it will effectively kill the system.
If it happens in a low priority task, higher priority tasks will continue to run, but lower priority tasks won't, so idle processing won't occur and depending on the task it may otherwise go unnoticed.

For example, I put an assert (counter < 5) in the blinky LED blink task, and the assertion happened and the LEDs stopped blinking, but the rest of the system appeared to be unaffected. Once the assert message scrolled off the screen, the only noticable thing was that the LEDs stopped blinking.

In a real app, we'd probably want to log this in a way that could be reported and then reset the system. If we were running under a debugger, I'd want it to drop into the debugger (i.e. make like you hit a breakpoint).

So I think we need some type of hook. I'm happy to have in a followup, but I think its needed.